src/issuance.rs

```diff
@@ -417,6 +438,45 @@ impl IssueBundle<Unauthorized> {
     }
 }
 
+impl<T: IssueAuth> IssueBundle<T> {
+    /// Returns the reference notes for the `IssueBundle`.
+    pub fn get_reference_notes(self) -> HashMap<AssetBase, Note> {
+        let mut reference_notes = HashMap::new();
+        self.actions.iter().for_each(|action| {
+            action.notes.iter().for_each(|note| {
+                if (note.recipient() == ReferenceKeys::recipient())
+                    && (note.value() == NoteValue::zero())
+                {
+                    reference_notes.insert(note.asset(), *note);
+                }
+            })
+        });
+        reference_notes
+    }
+}
+
+fn create_reference_note(asset: AssetBase, mut rng: impl RngCore) -> Note {
+    Note::new(
+        ReferenceKeys::recipient(),
+        NoteValue::zero(),
+        asset,
+        Rho::from_nf_old(Nullifier::dummy(&mut rng)),
+        &mut rng,
+    )
+}
+
+/// Validation for reference note
+///
+/// The following checks are performed:
+/// - the note value of the reference note is equal to 0
+/// - the asset of the reference note is equal to the provided asset
+/// - the recipient of the reference note is equal to the reference recipient
+pub fn verify_reference_note(note: &Note, asset: AssetBase) {
+    assert_eq!(note.value(), NoteValue::from_raw(0));
+    assert_eq!(note.asset(), asset);
+    assert_eq!(note.recipient(), ReferenceKeys::recipient());
+}
```

Comment:
Let's move it inside the test section. this code is not fit for real usage (no error handling)

Reply:
I used this function in tests/zsa.rs. Thus I cannot move it into test section.
Otherwise, I will have to duplicate it.

Reply:
I duplicate the function in order to be able to move it into test section.
I have no access to ReferenceKeys::recipeint() into tests/zsa.rs. Thus, I derived it from the zero SpendingKey.